contributing.md

```diff
@@ -209,6 +209,33 @@ When writing error messages, you should:
 
 See the <a href="{{site.baseurl}}/error-message-guidelines.html">error message guidelines</a> for more details.
 
+<h3>Behavior changes</h3>
+
+Behavior changes are user-visible functional changes in a new release via public APIs. The "user" here is not 
+only the user who writes queries and/or develops Spark plugins, but also the user who deploys and/or manages 
+Spark clusters. New features and bug fixes that eliminate `NullPointerException` or correct query results, are behavior 
+changes. However, performance improvement, code refactoring, and changes to unreleased APIs/features are 
+not.
+
+Everyone makes mistakes and so do Spark developers. We will continue to fix defects in Spark as they arise. However, it's important to communicate these behavior changes, so that Spark users can be 
+prepared during Spark version upgrades. If a PR introduces behavior changes, it should be called out 
+explicitly in the PR description. If the behavior change may require additional user actions, this should be called 
+out in the migration guide and where possible provide options to restore the previous behavior. Some examples:
+
+- Bug fixes that change query results. Users may need to backfill to correct the existing data and must
+know about these correctness fixes.
+- Bug fixes that change query schema. Users may need to update the schema of the tables in their data pipelines
+and must know about these changes.
+- Remove configs.
+- Rename error class/condition.
```

Comment:
Does changing error message itself count (not touch the error class name and condition, but rewording the error message)?

Reply:
no, error message change is OK (think about we support different languages in the future)